### PR TITLE
[core] Add High as a new warning level

### PR DESCRIPTION
**Pull request by tvandijck**
Maps to /W4 on VS and -Wall for gcc/clang removing -Wextra

### CODE REVIEW

File: tests/base/test_configset.lua [hunk 1]
```diff
@@ -177,14 +177,20 @@
 		local f = field.get("flags")
 		configset.store(cset, f, { "Symbols", "WinMain", "MFC" })
 		configset.remove(cset, f, { "WinMain" })
-		test.isequal({ "Symbols", "MFC" }, configset.fetch(cset, f, {}))
+
+		local result = configset.fetch(cset, f, {})
+		test.print(table.tostring(result))
```
Comment (samsinsane) [inline tests/base/test_configset.lua]: Debug code?

File: tests/base/test_configset.lua [hunk 2]
```diff
@@ -177,14 +177,20 @@
...
 	end
 
 	function suite.remove_onMultipleValues()
 		local f = field.get("flags")
 		configset.store(cset, f, { "Symbols", "Maps", "WinMain", "MFC" })
 		configset.remove(cset, f, { "Maps", "MFC" })
-		test.isequal({ "Symbols", "WinMain" }, configset.fetch(cset, f, {}))
+
+		local result = configset.fetch(cset, f, {})
+		test.print(table.tostring(result))
```
Comment (samsinsane) [inline tests/base/test_configset.lua]: Debug code?

File: tests/base/test_configset.lua (hunk 1 above)
Comment (tvandijck) [inline tests/base/test_configset.lua]: Yes, deliberately committed.. we very rarely see these two tests fail, and I want to have some extra output when it occurs.

Comment (samsinsane) [inline tests/base/test_configset.lua]: Sorry about that, I see the second commit message now.

Comment (tvandijck) [inline tests/base/test_configset.lua]: no worries

**Pull request merged**